manifests/apache.pp

```diff
@@ -26,15 +26,36 @@
 
   include ::apache::mod::headers
   include ::apache::mod::rewrite
+  include ::apache::mod::ssl
   apache::listen { $pimcore::params::port: }
 
-  $epp_config = { 'app_name' => $pimcore::app_name }
-  apache::vhost::custom { 'pimcore':
-    content => epp('pimcore/apache2.conf.epp', $epp_config)
-  }->
+  $epp_config = {
+    'app_name' => $pimcore::app_name,
+    'domain'   => $pimcore::params::domain
+  }
+
   file { "/var/www/html/${pimcore::app_name}":
     ensure  => 'link',
     target  => "/opt/pimcore/${pimcore::app_name}/public",
   }
+
+  apache::vhost { 'pimcore':
+    ssl         => true,
+    ssl_cert    => "/etc/letsencrypt/live/${pimcore::params::domain}/cert.pem",
+    ssl_key     => "/etc/letsencrypt/live/${pimcore::params::domain}/privkey.pem",
+    ssl_chain    => "/etc/letsencrypt/live/${pimcore::params::domain}/fullchain.pem",
```

Comment:
This has the effect of marrying the module to Letsencrypt. What I'd do is set these in ::pimcore::params, e.g.

    $ssl_cert = "/etc/letsencrypt/live/${::pimcore::params::domain}/cert.pem"

Then in the main class use those as defaults:

```
class pimcore (
   --- snip ---
  $ssl_cert = $::pimcore::params::ssl_cert
  --- snip ---
```
That way you're not required to use Letsencrypt certificates, but can do so easily.

I would also avoid parameters called $domain because they might accidentally get mixed up with the $::domain fact unless you specify the full path to the variable. E.g. using $domain in ::pimcore::params might have unintended consequences.